File: backend/models/BloqueSeleccion.py
from __future__ import annotations
from pydantic import BaseModel


class BloqueSeleccion(BaseModel):
    secciones: list[Seccion] = []
    bloquesLibres: list[BloqueTiempo] = []


from models.Seccion import Seccion
from models.BloqueTiempo import BloqueTiempo

File: src/models/BloqueSeleccion.ts
import Seccion from './Seccion';
import BloqueTiempo from './BloqueTiempo';

export default class BloqueSeleccion {
    secciones: Seccion[] = [];
    bloquesLibres: BloqueTiempo[] = [];

    constructor(secciones: Seccion[] = [], bloquesLibres: BloqueTiempo[] = []) {
        this.secciones = secciones;
        this.bloquesLibres = bloquesLibres;
    }
}
